### ISSUE TITLE
Cyclic dependency between DataFormats/TrackerCommon and DataFormats/SiStripDetId

### ISSUE THREAD

**Issue by smuzaffar** [cms-sw/cmssw#26776]
While building CXXMODULES IB, we noticed a cyclic dependency between DataFormats/TrackerCommon and DataFormats/SiStripDetId [a]. This can be avoided if we move TrackerTopology.* in to DataFormats/SiStripDetId itself or some other common package. If move to DataFormats/SiStripDetId sound reasonable then I can provide a PR.

[a]
```
>grep -r DataFormats/TrackerCommon DataFormats/SiStripDetId
DataFormats/SiStripDetId/src/SiStripSubStructure.cc:#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
DataFormats/SiStripDetId/BuildFile.xml:<use   name="DataFormats/TrackerCommon"/>
```
```
>grep -r DataFormats/SiStripDetId DataFormats/TrackerCommon
DataFormats/TrackerCommon/interface/ClusterSummary.h:#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
DataFormats/TrackerCommon/interface/TrackerTopology.h:#include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
DataFormats/TrackerCommon/interface/TrackerTopology.h:#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
DataFormats/TrackerCommon/src/TrackerTopology.cc:#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
```

**Comment by smuzaffar** [comment]
assign reconstruction

**Comment by smuzaffar** [comment]
assign simulation

**Comment by slava77** [comment]
It looks like SiStripSubStructure should be moved somewhere else.
Moving TrackerTopology to SiStrip* does not work from "first principles", because it has to serve all tracker variants, not just SiStrip.

**Comment by davidlange6** [comment]
I agree - likely to DataFormats/TrackerCommon?

**Comment by smuzaffar** [comment]
moving SiStripSubStructure in to DataFormats/TrackerCommon is even easier and involves less changes.,@perrotta  @slava77 @mdhildreth @civanch any objections?

**Comment by pieterdavid** [comment]
Another option would be to define the two enums that are needed from the headers in DataFormats/SiStripDetId to somewhere in DataFormats/TrackerCommon; out of
```
>grep -r DataFormats/SiStripDetId DataFormats/TrackerCommon
DataFormats/TrackerCommon/interface/ClusterSummary.h:#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
DataFormats/TrackerCommon/interface/TrackerTopology.h:#include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
DataFormats/TrackerCommon/interface/TrackerTopology.h:#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
DataFormats/TrackerCommon/src/TrackerTopology.cc:#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
```
the first is not needed, the second is only to use [this constructor](https://github.com/cms-sw/cmssw/blob/4985c11f34da6f6b475a04267946ea6aaa7a6bbe/DataFormats/SiStripDetId/interface/SiStripDetId.h#L31) (which could simply be replaced by the corresponding DetId one) and the [`SiStripDetId::ModuleGeometry`](https://github.com/cms-sw/cmssw/blob/4985c11f34da6f6b475a04267946ea6aaa7a6bbe/DataFormats/SiStripDetId/interface/SiStripDetId.h#L37) enum, and the third and fourth to pick up the [`StripSubdetector::SubDetector`](https://github.com/cms-sw/cmssw/blob/4985c11f34da6f6b475a04267946ea6aaa7a6bbe/DataFormats/SiStripDetId/interface/StripSubdetector.h#L13) enum.
In view of https://github.com/cms-sw/cmssw/issues/19398, using DataFormats/SiStripDetId is discouraged anyway, since most functionality is available from TrackerTopology in a more consistent and future-proof way.
Please let me know what you prefer (and, in the scenario above, where they could go. Maybe directly in TrackerTopology.h, as TrackerTopology::TIB or TrackerPartition::TIB etc., and SiStripModuleGeometry::IB1 etc? - I can prepare a PR for that). If you want a quick solution, moving more to TrackerCommon should work, and we can reorganise what (currently) remains of DataFormats/SiStripDetId later on.

**Comment by slava77** [comment]
@pieterdavid 
you (Tracker DPG) are perhaps the main customer for this.
If you could follow up soon with a PR that fits your needs and also resolves the cyclic dependency, it would be nice.
Thank you.

**Comment by slava77** [comment]
+1

I think that this issue was resolved with #26804, which was merged recently.
